pjrt/convert/trace_container.zig

```diff
@@ -0,0 +1,239 @@
+const std = @import("std");
+const trace_events_proto = @import("//tsl:trace_events_proto");
+const xplane_proto = @import("//tsl:xplane_proto");
+const xplane_schema = @import("xplane_schema.zig");
+const xplane_visitor = @import("xplane_visitor.zig");
+
+// Constants used as trace_viewer PID (device_id in trace_events.proto).
+// PID 0 is unused.
+// Support up to 500 accelerator devices.
+const first_device_id = 1;
+const last_device_id = 500;
+// Support Upto 200 custom planes as fake devices (i.e., planes with a
+// "/custom:" prefix). See `<project_name>::custom_plane_prefix` for more
+// information
+const first_custom_plane_device_id = last_device_id + 1;
+const max_custom_plane_devices_per_host = 200;
+const last_custom_plane_device_id = first_custom_plane_device_id + max_custom_plane_devices_per_host - 1;
+
+// Host threads are shown as a single fake device.
+pub const host_threads_device_id = last_custom_plane_device_id + 1;
+
+pub const xla_async_op_line_name = "Async XLA Ops";
+
+pub const host_threads_plane_name = "/host:CPU";
+pub const gpu_plane_prefix = "/device:GPU:";
+pub const tpu_plane_prefix = "/device:TPU:";
+pub const custom_plane_prefix = "/device:CUSTOM:";
+
+pub const TraceContainer = struct {
+    metadata: Trace = .{},
+    events: std.ArrayListUnmanaged(*TraceEvent) = .{},
+
+    pub const Trace = struct {
+        devices: std.AutoArrayHashMapUnmanaged(u32, Device) = .{},
+        trace_events: std.ArrayListUnmanaged(TraceEvent) = .{},
```

Comment:
I don't think this field is used at all ?

* if it's not used: please remove it, and replace `metadata.devices` by just `devices`
* if it's used: why do you have both `events` and `metadata.trace_events` ?